bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookieWatcher.java

```diff
@@ -363,14 +363,18 @@ public void processResult(int rc, String path, Object ctx, List<String> children
 
         void notifyBookiesChanged(final BookiesListener listener) throws BKException {
             try {
-                bk.getZkHandle().getChildren(this.readOnlyBookieRegPath, new Watcher() {
+                List<String> children = bk.getZkHandle().getChildren(this.readOnlyBookieRegPath, new Watcher() {
                     public void process(WatchedEvent event) {
                         // listen children changed event from ZooKeeper
                         if (event.getType() == EventType.NodeChildrenChanged) {
                             listener.availableBookiesChanged();
                         }
                     }
                 });
+
+                // Update the list of read-only bookies
+                HashSet<BookieSocketAddress> newReadOnlyBookies = convertToBookieAddresses(children);
+                readOnlyBookies = newReadOnlyBookies;
```

Comment:
Yes, it's exactly the same. I just left it in the same form it was already written at line 404 in the same file